File: cs471/HW/hw1/ack.c
#include <stdio.h>

int ack(int m, int n) {
	if (!m) return n + 1;
    if (!n) return ack(m - 1, 1);
    return ack(m - 1, ack(m, n - 1));
}

int main(int argc, char **argv) {
  if (argc < 3) {
    printf("%s usage: [m] [n]\n", argv[0]);
    return 1;
  }
  int m = atoi(argv[1]);
  int n = atoi(argv[2]);
  int r = ack(m,n);
  
  printf("%d\n", r);

  return 0;
}

File: cs471/HW/hw1/ack.py
#! /usr/bin/env python
import sys

def ack(m, n):
	if m == 0:
		return (n+1)
	elif n == 0:
		return ack(m-1, 1)
	else:
		return ack(m-1, ack(m, n-1))

if len(sys.argv) < 3:
  print("%s usage: [I] [J]" % sys.argv[0])
  exit()

print(ack(int(sys.argv[1]), int(sys.argv[2])))